app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,44 @@
+class CategoriesController < ApplicationController
+  include CategoriesHelper
+
+  before_action :set_category, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @categories = Category.all
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.create(category_params)
+    create_helper
```

Comment:
we already spoke about how this is probably extracting too much logic out of our views. some things we can keep in our controller methods 👍